### PR TITLE
Validate MSVC Windows build 

### PR DESCRIPTION
**Pull request by AndreaLanfranchi**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -53,7 +53,10 @@ target_include_directories(secp256k1 PRIVATE secp256k1 INTERFACE secp256k1/inclu
 
 # Compiler options
 if(MSVC)
-  add_compile_options(/W4 /WX)
```
Comment (yperbasis) [inline CMakeLists.txt:56]: Shall we keep /WX?

File: CMakeLists.txt [hunk 2]
```diff
@@ -87,7 +90,13 @@ target_link_libraries(evmone PUBLIC evmc intx::intx PRIVATE ethash::keccak)
 add_subdirectory(silkworm)
 
 add_executable(check_changes cmd/check_changes.cpp)
-target_link_libraries(check_changes PRIVATE silkworm absl::flags_parse absl::time)
+
+if(MSVC)
+  # lmdb requires link to ntdll.lib
```
Comment (yperbasis) [inline CMakeLists.txt:95]: Let's link silkworm to ntdll rather than check_changes and tests individually

Comment (AndreaLanfranchi) [reply CMakeLists.txt:95]: Wilco

File: CMakeLists.txt (hunk 1 above)
Comment (AndreaLanfranchi) [reply CMakeLists.txt:56]: Yes we can but have to resolve several warnings

Comment (yperbasis) [reply CMakeLists.txt:56]: I see

**Pull request merged**